channeld/channeld.c

```diff
@@ -2915,6 +2919,20 @@ static void init_channel(struct peer *peer)
 		     feerate_per_kw[LOCAL], feerate_per_kw[REMOTE],
 		     peer->feerate_min, peer->feerate_max);
 
+	if(remote_ann_node_sig && remote_ann_bitcoin_sig) {
+		peer->announcement_node_sigs[REMOTE] = *remote_ann_node_sig;
+		peer->announcement_bitcoin_sigs[REMOTE] = *remote_ann_bitcoin_sig;
+		peer->have_sigs[REMOTE] = true;
+
+		tal_steal(tmpctx, remote_ann_node_sig);
+		tal_steal(tmpctx, remote_ann_bitcoin_sig);
```

Comment:
Why are you stealing these onto `tmpctx`? You could also pass the pointers to `peer->announcement_node_sigs[REMOTE] ` and `peer->announcement_bitcoin_sigs[REMOTE]` into the `fromwire_channel_init` call directly, which avoids having these temp variables and having to deal with intermediate allocations and ownership.

Reply:
Yes, I find these tal_steal are extra. But I think `fromwire_channel_init` needs these 2 pointer, because they can handle `NULL` sigs:

> `channel_init,,remote_ann_node_sig,?secp256k1_ecdsa_signature` `channel_init,,remote_ann_bitcoin_sig,?secp256k1_ecdsa_signature`

Reply:
emm, after deleting tal_steal ,Travis CI tell me memory leak. So I tal_free these directly.(I find other variables used in fromwire_channel_init are freed directly, so I also do it.)